--- libcwiid/bluetooth.c	2007-04-24 22:41:13.000000000 -0400
+++ libcwiid/bluetooth.c.oden	2008-11-18 11:48:26.000000000 -0500
@@ -122,7 +122,7 @@ int cwiid_get_bdinfo_array(int dev_id, u
 		}
 
 		/* timeout (10000) in milliseconds */
-		if (hci_remote_name(sock, &dev_list[i].bdaddr, BT_NAME_LEN,
+		if (hci_read_remote_name(sock, &dev_list[i].bdaddr, BT_NAME_LEN,
 		                    (*bdinfo)[bdinfo_count].name, 10000)) {
 			cwiid_err(NULL, "Bluetooth name read error");
 			err = 1;
